code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -8,6 +8,11 @@ This prevents race conditions that arise based on the order of tile processing.
 GLOBAL_LIST_INIT(meta_gas_info, meta_gas_list()) //see ATMOSPHERICS/gas_types.dm
 GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
+#define ATMOS_GARBAGE_COLLECT(gaslist, tocheck)\
+	for(var/datum/gas/id as anything in (tocheck || gaslist)){\
+		if(QUANTIZE(gaslist[id][MOLES]) <= 0) gaslist -= id;\
+	}
+
```

Comment:
This isn't a valid change. Garbage collect is overriden by code that needs to be able to react to it.
We cannot move it to a macro